collectors/cgroups.plugin/sys_fs_cgroup.c

```diff
@@ -1862,8 +1815,149 @@ static inline void cgroup_free(struct cgroup *cg) {
+    debug(D_CGROUP, "executing command %s \"%s\" for cgroup '%s'", cgroups_rename_script, cg->intermediate_id, cg->chart_id);
+    pid_t cgroup_pid;
+
+    FILE *fp;
+    (void)mypopen_raw_default_flags_and_environment(&cgroup_pid, &fp, cgroups_rename_script, cg->id, cg->intermediate_id);
+    if (!fp) {
+        error("CGROUP: cannot popen(%s \"%s\", \"r\").", cgroups_rename_script, cg->intermediate_id);
+        cg->pending_renames = 0;
+        cg->processed = 1;
+        return;
+    }
+
+    char buffer[CGROUP_CHARTID_LINE_MAX + 1];
+    char *new_name = fgets(buffer, CGROUP_CHARTID_LINE_MAX, fp);
+    int exit_code = mypclose(fp, cgroup_pid);
+
+    switch (exit_code) {
+        case 0:
+            cg->pending_renames = 0;
+            break;
+        case 3:
+            cg->pending_renames = 0;
+            cg->processed = 1;
+            break;
+        default:
+            if (!cg->pending_renames && is_inside_k8s) {
+                cg->processed = 1;
+            }
+    }
+
+    if (cg->pending_renames || cg->processed) {
+        return;
+    }
+    if (!(new_name && *new_name && *new_name != '\n')) {
+        return;
+    }
+    new_name = trim(new_name);
+    if (!(new_name)) {
+        return;
+    }
+    char *name = new_name;
+    if (!strncmp(new_name, "k8s_", 4)) {
+        free_label_list(cg->chart_labels);
+        name = k8s_parse_resolved_name(&cg->chart_labels, new_name);
+    }
+    freez(cg->chart_title);
+    cg->chart_title = cgroup_title_strdupz(name);
+    freez(cg->chart_id);
+    cg->chart_id = cgroup_chart_id_strdupz(name);
+    substitute_dots_in_id(cg->chart_id);
+    cg->hash_chart = simple_hash(cg->chart_id);
+}
+
+static void is_cgroup_procs_exist(netdata_ebpf_cgroup_shm_body_t *out, char *id) {
+    struct stat buf;
+
+    snprintfz(out->path, FILENAME_MAX, "%s%s/cgroup.procs", cgroup_cpuset_base, id);
+    if (likely(stat(out->path, &buf) == 0)) {
+        return;
+    }
+
+    snprintfz(out->path, FILENAME_MAX, "%s%s/cgroup.procs", cgroup_blkio_base, id);
+    if (likely(stat(out->path, &buf) == 0)) {
+        return;
+    }
+
+    snprintfz(out->path, FILENAME_MAX, "%s%s/cgroup.procs", cgroup_memory_base, id);
+    if (likely(stat(out->path, &buf) == 0)) {
+        return;
+    }
+
+    snprintfz(out->path, FILENAME_MAX, "%s%s/cgroup.procs", cgroup_devices_base, id);
+    if (likely(stat(out->path, &buf) == 0)) {
+        return;
+    }
+
+    out->path[0] = '\0';
+    out->enabled = 0;
+}
+
+static inline void convert_cgroup_to_systemd_service(struct cgroup *cg) {
+    char buffer[CGROUP_CHARTID_LINE_MAX];
+    cg->options |= CGROUP_OPTIONS_SYSTEM_SLICE_SERVICE;
+    strncpyz(buffer, cg->id, CGROUP_CHARTID_LINE_MAX);
+    char *s = buffer;
+
+    // skip to the last slash
+    size_t len = strlen(s);
+    while (len--)
+        if (unlikely(s[len] == '/'))
+            break;
+    if (len)
+        s = &s[len + 1];
+
+    // remove extension
+    len = strlen(s);
+    while (len--)
```

Comment:
put braces here

Reply:
@ilyam8 I already tested your PR on Linux distributions that uses `cgroup v1` and `cgroup v2`, I am only waiting a commit or comments about this to approve it.

Reply:
done